### PR TITLE
Fixes #33359 - move client_module_class to repo type registry

### PR DESCRIPTION
**Pull request by jlsherrill**
for all types

### CODE REVIEW

**Comment by jlsherrill** [comment]
ended up moving the rest as part of a 2nd commit, we can separate if desired

File: lib/katello/repository_types/yum.rb [hunk 1]
```diff
@@ -5,6 +5,20 @@
   pulp3_plugin 'rpm'
   prevent_unneeded_metadata_publish
 
+  client_module_class PulpRpmClient
+  remote_class PulpRpmClient::RpmRpmRemote
+  client_module_class PulpRpmClient
```
Comment (ianballou) [inline lib/katello/repository_types/yum.rb:10]: `client_module_class` is listed twice.

File: lib/katello/repository_types/docker.rb [hunk 2]
```diff
@@ -8,6 +8,18 @@
 
   set_unique_content_per_repo
 
+  client_module_class PulpContainerClient
```
Comment (ianballou) [inline lib/katello/repository_types/docker.rb:11]: The api classes for Docker seem to be ordered a bit differently from the rest.

**Review by ianballou — COMMENTED** [review]
Did some more testing, everything seems fine here. Just my first comments up above.

**Comment by jlsherrill** [comment]
I reordered them a bit to better group things that made sense

**Review by ianballou — APPROVED** [review]
Looks good!

**Pull request merged**